docs/advanced.rst

```diff
@@ -562,3 +562,79 @@ or the record invocation: ``app="SomeAppName"``.
         opinion = models.CharField(max_length=2000)
 
     register(Opinion, app="SomeAppName")
+
+Custom History Model Name
+-------------------------
+
+By default the history model names are the base model name prefixed with "Historical".
+So given a model named ``Poll``, the resultant history class will be named ``HistoricalPoll``.
+If the app name of the ``Poll`` model is `election` then the history table name would be ``election_historicalpoll``.
+
+This feature allows a different and/or shorter model name and would even allow model names to be in the standard Django form
+
+    [appName]_[modelName]
+
+where the history models are created in a dedicated app using the "app" option thus making the resultant history table
+name much more aligned with the base table it is tracking
+
+To configure history models to use a different name for the history model class, use an option named ``custom_model_name``
+
+The value for this option can be a string or a callable.
+
+The most simple use case is illustrated below:
+
+.. code-block:: python
+
+    class ModelNameExample(models.Model):
+        history = HistoricalRecords(
+            custom_model_name='SimpleHistoricalModelNameExample'
+        )
+
+As an example using the callable mechanism, the below changes the default prefix `Historical` to `Audit`:
+
+.. code-block:: python
+
+    class Poll(models.Model):
+        question = models.CharField(max_length=200)
+        history = HistoricalRecords(custom_model_name=lambda x:f'Audit{x}')
+
+    class Opinion(models.Model):
+        opinion = models.CharField(max_length=2000)
+
+    register(Opinion, custom_model_name_prefix=lambda x:f'Audit{x}')
+    
+The resulting history class names would be `AuditPoll` and `AuditOpinion`.
+If the app the models are defined in is `yoda` then the corresponding history table names would be `yoda_auditpoll` and `yoda_auditopinion`
+ 
+You could also use a base class for your models and declare the history instantiation in the base class.
+All models that inherit the base model would automatically get this custom name.
+Using this mechanism requires using the `inherit` option.
+
+.. code-block:: python
+
+    class BaseModel(models.Model):
+        history = HistoricalRecords(inherit=True, app="yoda", custom_model_name_prefix=lambda x:f'Audit{x}')
+        class Meta:
+            abstract = True
+
+    class Poll(BaseModel):
+        question = models.CharField(max_length=200)
+
+    class Opinion(BaseModel):
+        opinion = models.CharField(max_length=2000)
+
+The resulting history class names would be `AuditPoll` and `AuditOpinion`.
```

Comment:
Not sure you need this section on inheritance, but if you want to keep it, I think you'd be better off creating a subclass of `HistoricalRecords`. For example:
```
class NewHistoricalRecords(HistoricalRecords):
    def __init__(*args, **kwargs):
        kwargs['custom_model_name'] = lambda x: f'Audit{x}'
        kwargs['app'] = 'yoga'
        super().__init__(*args, **kwargs)
```

Reply:
Would that not require you put the instantiation in all the concrete class then? So would end up like this:

class NewHistoricalRecords(HistoricalRecords):
    def __init__(*args, **kwargs):
        kwargs['custom_model_name'] = lambda x: f'Audit{x}'
        kwargs['app'] = 'yoga'
        super().__init__(*args, **kwargs)

    class Poll(BaseModel):
        question = models.CharField(max_length=200)
        history = NewHistoricalRecords()

    class Opinion(BaseModel):
        opinion = models.CharField(max_length=2000)
       history = NewHistoricalRecords()

Reply:
Thought the goal was to have a consistent naming convention with the same app for history, not just inheritance

Reply:
From my own point of view the goal was to be able to keep the table name short and in the same format as the tracked table but in a different app. The classname for the historical model is not really of interest to me.

Reply:
Then this seems like a roundabout way of getting there. If we're going to use a custom_model_name, I'd prefer that it's used for the actual model name as opposed for the table name. Maybe you're better off allowing the `table_name` parameter be a callable?

Reply:
Agreed but it seemed like the right way to do it since the classname by default defines the table name so figured it would be best to try to "stay within the reservation" as it were.
I can remove setting the table name and simply throw an error if the class name for the historical model would be the same as the model it is tracking and then for that special case add the callable on "table_name" in a separate piece of work?

Reply:
So you'd make it such that the historical model name can be anything except for the current model name? And do a separate PR for a custom table name? That's ok with me

Reply:
So shall I just add to the current PR or finish this and start another based on a separate issue I create?

Reply:
Let's keep them separate

Reply:
Also remember to rebase!